streampark-flink/streampark-flink-shims/streampark-flink-shims-base/src/main/scala/org/apache/streampark/flink/core/SqlCommandParser.scala

```diff
@@ -172,6 +172,19 @@ object SqlCommand extends enumeratum.Enum[SqlCommand] {
         "create catalog",
         "(CREATE\\s+CATALOG\\s+.+)")
 
+  /**
+   * <pre>
+   * WITH <with_item_definition> [ , ... ]
+   * SELECT ... FROM ...;
+   *
+   * <with_item_defintion>:
+   * with_item_name (column_name[, ...n]) AS ( <select_query> )
+   * </pre>
+   */
+  case object WITH extends SqlCommand(
+      "with",
+      "(WITH.+AS.+\\(\\s*SELECT.+\\))")
```

Comment:
can you provide test case? this code can pass? `withas (selectxxx)` 
if not passed, You can test it again with this regexp:
```"(WITH\\s+.+AS\\s+.+\\(\\s*SELECT\\s+.+\\))")```